Library/Formula/qt.rb

```diff
@@ -25,6 +25,9 @@ class Qt < Formula
     sha256 "c8a0fa819c8012a7cb70e902abb7133fc05235881ce230235d93719c47650c4e"
   end
 
+  # Make `HOMEBREW_PREFIX/lib/qt4/plugins` an additional plugin search path.
+  patch :DATA
```

Comment:
> Presumably Qt5 will need a similar fix?

I'm not sure and haven't checked if there is now a more straightforward way to have multiple predefined plug-in directories. Plus I'm not aware of any formulae that provide plug-ins for `qt5`.

> […] I'm happy just reducing functionality (e.g. plugins) rather than killing it completely.

Given that the only application really interested in those plug-ins is Qt Designer (all affected formulae install Designer plug-ins), I have one more idea of how to resolve this with relative elegance and without source/binary patching. I'll check if it will work like I think it should.